### PR TITLE
More consistent behavior between Mesh and Non Mesh stealing

### PR DESCRIPTION
**Pull request by DmitryDodzin**
This changes when tcp stealing on a meshed service, requests from localhost to localhost like healthprobes will skip mirrord 
and will be routed to the underlying service like when stealing from a non meshed service

Relevant for #1044

### CODE REVIEW

File: mirrord/agent/src/main.rs [hunk 1]
```diff
@@ -555,14 +555,14 @@ async fn start_iptable_guard() -> Result<()> {
 
     let result = spawn_child_agent();
 
-    run_thread_in_namespace(
+    let _ = run_thread_in_namespace(
```
Comment (meowjesty) [inline mirrord/agent/src/main.rs:558]: Aren't we eating up the error now with this?

File: mirrord/agent/src/steal/ip_tables.rs [hunk 2]
```diff
@@ -220,24 +226,38 @@ where
...
 impl IPTableFormatter {
     const MESH_OUTPUTS: [&'static str; 2] = ["-j PROXY_INIT_OUTPUT", "-j ISTIO_OUTPUT"];
+    const MESH_NAMES: [&'static str; 2] = ["PROXY_INIT_OUTPUT", "ISTIO_OUTPUT"];
 
     #[tracing::instrument(level = "trace", skip_all)]
     pub(crate) fn detect<IPT: IPTables>(ipt: &IPT) -> Result<Self> {
         let output = ipt.list_rules("OUTPUT")?;
 
-        if output.iter().any(|rule| {
+        if let Some(mesh_ipt_chain) = output.iter().find_map(|rule| {
```
Comment (meowjesty) [inline mirrord/agent/src/steal/ip_tables.rs:243]: Probably good to add a comment here mentioning that we need this for the health probes.

**Review by meowjesty — CHANGES_REQUESTED** [review]
Just requesting some docs, and I think you're eating up an error.

File: mirrord/agent/src/main.rs (hunk 1 above)
Comment (aviramha) [reply mirrord/agent/src/main.rs:558]: It's by design since it can fail

**Review by aviramha — APPROVED** [review]
LGTM (besides meow comments)

**Review by meowjesty — APPROVED** [review]
:bear: :+1:

**Pull request merged**